src/memop.rs

```diff
@@ -3,16 +3,19 @@
 use crate::syscalls;
 use core::slice;
 
-// Setting the break is marked as unsafe as it should only be called by the entry point to setup
-// the allocator. Updating the break afterwards can lead to allocated memory becoming unaccessible.
-//
-// Alternate allocator implementations may still find this useful in the future.
+/// Set the memory break
+/// # Safety
+/// Setting the break is marked as unsafe as it should only be called by the entry point to setup
+/// the allocator. Updating the break afterwards can lead to allocated memory becoming unaccessible.
+///
+/// Alternate allocator implementations may still find this useful in the future.
 pub unsafe fn set_brk(ptr: *const u8) -> bool {
     syscalls::raw::memop(0, ptr as usize) == 0
 }
 
-pub unsafe fn increment_brk(increment: usize) -> Option<*const u8> {
-    let result = syscalls::raw::memop(1, increment);
+/// Increment the memory break
+pub fn increment_brk(increment: usize) -> Option<*const u8> {
```

Comment:
Is making this `safe` intentional? Although with the current allocator this shouldn't be problematic (this function isn't used), in general if we want to be more generic in terms of allocators (in particular supporting break-aware allocators), this would be unsafe, and up to the allocator's implementation to prove that it is indeed safe.

Reply:
I would leave this `safe` as long we don't allow alternative allocators.

Reply:
On the other hand, the current allocator doesn't use it, so there's no reason to use this function in the current setup. Therefore I would make it `unsafe`.

Also, I don't see a reason to forbid alternative allocators. Of course, up to anyone willing to try other allocators to make that work, but I don't think libtock-rs should make it hard to swap the allocator implementation.

Reply:
Would you mind clarifying how increment_brk can break memory safety? It's not obvious to me that it is unsafe.